### PR TITLE
Add NETWORK_NAME_or_ID value for --net= option

### PR DESCRIPTION
**Pull request by wenchma**
Signed-off-by: Wen Cheng Ma wenchma@cn.ibm.com

### CODE REVIEW

**Comment by vdemeester** [comment]
ping @thaJeztah @moxiegirl

File: docs/reference/commandline/run.md [hunk 1]
```diff
@@ -58,8 +58,8 @@ parent = "smn_cli"
                                     'bridge': creates a new network stack for the container on the docker bridge
                                     'none': no networking for this container
                                     'container:<name|id>': reuses another container network stack
-                                    'host': use the host network stack inside the container
-                                    'NETWORK': connects the container to user-created network using `docker network create` command
+                                    'host': uses the host network stack inside the container
+                                    'NETWORK_NAME_or_ID': connects the container to user-defined network using `docker network create` command
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md]: I'm a bit in doubt if we should be this explicit, or just use "NETWORK", and mention that either the name or id of the network can be used.

@moxiegirl wdyt?

Comment (thaJeztah) [inline docs/reference/commandline/run.md]: for reference, the `docker network rm` command currently also uses `NETWORK` but mentions that both "name" and "id" can be used; https://github.com/docker/docker/blob/master/docs/reference/commandline/network_rm.md

**Comment by thaJeztah** [comment]
Thanks for your contribution @wenchma!

While looking, I also noticed that the documentation and man-page for "create" is missing this information, perhaps (once we agree on the right wording), we can update that as well in this PR; https://github.com/docker/docker/blob/master/docs/reference/commandline/create.md

File: docs/reference/commandline/run.md (hunk 1 above)
Comment (wenchma) [inline docs/reference/commandline/run.md]: @thaJeztah for https://github.com/docker/docker/blob/master/docs/reference/commandline/network_rm.md ,  the name and id have been mentioned. also need to change ?

**Comment by wenchma** [comment]
@thaJeztah thanks for your comments, :+1:   all are addressed, please check the current PR.

**Comment by wenchma** [comment]
@moxiegirl any comments on the current PR?

File: docs/reference/commandline/create.md [hunk 2]
```diff
@@ -55,7 +55,12 @@ Creates a new container.
       --memory-swap=""              Total memory (memory + swap), '-1' to disable swap
       --memory-swappiness=""        Tune a container's memory swappiness behavior. Accepts an integer between 0 and 100.
       --name=""                     Assign a name to the container
-      --net="default"               Set the Network mode for the container
+      --net="bridge"                Connects a container to a network
+                                    'bridge': creates a new network stack for the container on the docker bridge
+                                    'none': no networking for this container
+                                    'container:<name|id>': reuses another container network stack
+                                    'host': uses the host network stack inside the container
+                                    'NETWORK': connects the container to user-defined network using `docker network create` command, either the name or id can be used.
```
Comment (thaJeztah) [inline docs/reference/commandline/create.md]: looking at this again, perhaps we should use a similar approach as `container:`;

```Markdown
'<network-name|network-id>': connects the container to a user-defined network.
```

wdyt?

Comment (wenchma) [inline docs/reference/commandline/create.md]: @thaJeztah good idea, is it the same as others docs ?

Comment (thaJeztah) [inline docs/reference/commandline/create.md]: Yes, if you can change it in all the docs (that are in this PR), I think that'd be great.

Comment (moxiegirl) [inline docs/reference/commandline/create.md]: @wenchma Thank you for the contribution.  I think there are few things in this that need to be tighten.  For example, `create new` is redundant as create implies something didn't exist before.  Omit phrases that refer back to the container --- the intro sentence makes the unnecessary. Also, we should resist the temptation to provide too much information in the option --- for the `NETWORK` option the description is sliding into user info.  We should rely on the description of the examples  for that --- there are some in Examples for connecting to a network.  

This brings it closer:

```markdown
--net="bridge"         Connect a container to a network
                              'bridge': create a network stack on the default Docker bridge
                              'none': no networking
                              'container:<name|id>': reuse another container's network stack
                              'host': use the Docker host network stack
                              'NETWORK': connect to a user-defined network
```

File: docs/userguide/networking/default_network/container-basics.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ The steps with which Docker configures a container are:
 
 With these steps complete, the container now possesses an `eth0` (virtual) network card and will find itself able to communicate with other containers and the rest of the Internet.
 
-You can opt out of the above process for a particular container by giving the `--net=` option to `docker run`, which takes four possible values.
+You can opt out of the above process for a particular container by giving the `--net=` option to `docker run`, which takes five possible values.
```
Comment (moxiegirl) [inline docs/userguide/networking/default_network/container-basics.md]: `which takes these values:`   is cleaner and requires less maintenance than counting

Comment (wenchma) [inline docs/userguide/networking/default_network/container-basics.md]: @moxiegirl good idea :+1:

File: docs/reference/commandline/create.md (hunk 2 above)
Comment (wenchma) [inline docs/reference/commandline/create.md]: @moxiegirl  done

**Comment by wenchma** [comment]
@moxiegirl @thaJeztah ,  much appreciate your comments, addressed, please have a review again, thanks. :+1:

**Comment by thaJeztah** [comment]
Thank you @wenchma, and thanks for being patient with us :-)

docs LGTM

ping @moxiegirl PTAL

**Comment by thaJeztah** [comment]
aw, looks like this needs a rebase

**Comment by wenchma** [comment]
rebased

**Comment by thaJeztah** [comment]
Oh my, it really is the day for rebases, because it already needs another rebase. (Can't believe it).

**Comment by moxiegirl** [comment]
@wenchma Thank you for your patience and quickness. LGTM

**Comment by thaJeztah** [comment]
merging, this PR doesn't change code, so ignoring users CI

**Comment by thaJeztah** [comment]
Thanks @wenchma!

**Pull request merged**